ext/json/json.c

```diff
@@ -178,6 +178,28 @@ PHP_JSON_API zend_result php_json_decode_ex(zval *return_value, const char *str,
 }
 /* }}} */
 
+/* {{{ */
+PHP_JSON_API zend_result php_json_validate_ex(zval *return_value, const char *str, size_t str_len, zend_long options, zend_long depth)
+{
+	php_json_parser parser;
+
+	php_json_parser_init(&parser, return_value, str, str_len, (int)options, (int)depth, true);
+
+	if (php_json_yyparse(&parser)) {
+		php_json_error_code error_code = php_json_parser_error_code(&parser);
+		if (!(options & PHP_JSON_THROW_ON_ERROR)) {
+			JSON_G(error_code) = error_code;
```

Comment:
A validation routine should not have side effects, such as setting a global `error_code`.

Reply:
I will check this and will be back to you

Reply:
The reason I kept that code, was because of the usage of json_last_error() function. 

I , and other developers in the mailing list find it useful to have such information, to know why a validation failed so we can have messages like:

- Malformed UTF-8 characters, possibly incorrectly encoded
- Syntax error
- Maximum stack depth exceeded

etc.

I think is good thing to have, also some other devs from mailing list.

What do you think?

Reply:
I think this is a legit solution; an alternative could be to return the error code as int (instead of a bool).

Reply:
I wrote this down in the RFC as an open question